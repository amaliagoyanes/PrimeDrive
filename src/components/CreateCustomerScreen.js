import React, {Component} from 'react';
import {
    TouchableOpacity,
    Text,
    Alert,
    Image,
    View
} from 'react-native';
import {FormattedMessage} from 'react-intl';
import {Col, Row, Grid} from 'react-native-easy-grid';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
import {Sae} from 'react-native-textinput-effects';
import {Actions} from 'react-native-router-flux';
import {connect} from 'react-redux';
import SelectMultiple from 'react-native-select-multiple';
import ImagePicker from 'react-native-image-picker';
import firestack from '../fbconfig.js'
import imageUpload from '../imageUpload.js'

import {ButtonItem, guid4, AlertPopup} from './index';
import styles from '../style/styles';
import {scale, scaleText} from '../scaling';
import ReceiptListViewRow from "./plain/ReceiptListViewRow";

class CreateCustomerScreen extends Component {
    constructor(props) {
        super(props);

        this.state = {
            firstName: '', lastName: '', address: '', city: '', zipcode: '', accountNumber: '', mail: '',
            phone: '', notes: '', customerGroups: [], groups: [], uid: '', imageURL: '', customerCreated: false,
            accountName: '', accountCardID: '', accountCardType: '', accountVoucher: '', accountTypes: [],
            accountType: [], accountCredit: '', accountBalance: ''
        };

        this.okOnPress = this.okOnPress.bind(this);
        this.getGroups = this.getGroups.bind(this);
        this.onSelectionsChange = this.onSelectionsChange.bind(this);
        this.pickImage = this.pickImage.bind(this);
        this.createCustomer = this.createCustomer.bind(this);
        this.createCustomerAccount = this.createCustomerAccount.bind(this);
        this.onAccountTypeChange = this.onAccountTypeChange.bind(this);
    }

    componentDidMount() {
        this.getGroups();
        this.getAccountTypes();
        this.setState({uid: guid4() + guid4() + '-' + guid4() + guid4() + guid4()});
    }

    getGroups() {
        firestack.database.ref(`/customerGroups/`).once("value").then((snapshot) => {
            let groups = snapshot.val();
            groups = Object.keys(groups).map(group => {
                return {label: groups[group].name, value: group}
            });
            this.setState({groups: groups})
        })
    }

    getAccountTypes() {
        firestack.database.ref(`/customerAccountsType/`).once("value").then((snapshot) => {
            let types = snapshot.val();
            types = Object.keys(types).map(type => {
                return {label: types[type].name, value: type}
            });
            this.setState({accountTypes: types})
        })
    }

    createCustomer() {
        if (!this.state.firstName.length) {
            this._firstName.focus()
        } else if (!this.state.lastName.length) {
            this._lastName.focus()
        } else if (!this.state.accountNumber.length) {
            this._accountNumber.focus()
        }
        else {
            let groups = {};
            this.state.customerGroups.forEach(function (group) {
                groups[group.value] = true;
            });
            let customer = {
                name: this.state.firstName + ' ' + this.state.lastName,
                address: this.state.address,
                city: this.state.city,
                zipcode: this.state.zipcode,
                accountNumber: this.state.accountNumber,
                mail: this.state.mail,
                phone: this.state.phone,
                notes: this.state.notes,
                groups: groups,
                image: this.state.imageURL
            };
            firestack.database.ref(`/customers/${this.state.uid}`).set(customer).then((snapshot) => {
                this.setState({customerCreated: true})
            }).catch((err) => {
                console.log('Error', err)
            });
        }
    }

    createCustomerAccount() {
        if (!this.state.accountName.length) {
            this._accountName.focus()
        }else if (!this.state.accountCardID.length) {
            this._accountCardID.focus()
        } else if (!this.state.accountCardType.length) {
            this._accountCardType.focus()
        } else if (!this.state.accountCredit.length) {
            this._accountCredit.focus()
        }else if (!this.state.accountBalance.length) {
            this._accountBalance.focus()
        }else if (!this.state.accountType.length) {
            this._alert.open('Please select account type!')
        } else {
            let account = {
                name: this.state.accountName,
                voucher_code: this.state.address,
                active: true,
                credit: this.state.accountCredit,
                balance: this.state.accountBalance,
                card_id: this.state.accountCardID,
                card_type: this.state.accountCardType,
                customer_id: this.state.uid,
                account_type: this.state.accountType[0].value
            };
            let accountUid = guid4() + guid4() + guid4() + guid4() + guid4();
            firestack.database.ref(`/customerAccounts/${accountUid}`).set(account).then((snapshot) => {
                this._alert.open('Success');
            }).catch((err) => {
                console.log(err);
            });
        }
    }

    okOnPress() {
        if (!this.state.customerCreated) this.createCustomer();
        else this.createCustomerAccount();
    }

    onSelectionsChange(customerGroups) {
        this.setState({customerGroups})
    };

    onAccountTypeChange(accountType) {
        {/* hack: there's no good component for select dropdown */}
        if (accountType.length > 1) {
            this.setState({accountType: accountType.slice(1)})
        }
        else this.setState({accountType})
    };

    pickImage() {
        this.setState({imageURL: ''});

        ImagePicker.launchImageLibrary({}, response => {
            imageUpload(response.uri, this.state.uid, '/customers/images')
                .then(url => this.setState({imageURL: url}))
                .catch(error => console.log(error))
        })
    }

    render() {
        let header = {
            height: scale(40),
            alignSelf: 'stretch',
        };

        return (
            <Grid style={[styles.container, {paddingLeft: scale(30), paddingTop: scale(25)}]}>
                <Row size={0.5} style={header}>
                    <Col size={2}>
                        <Text style={{color: "white", marginTop: 5}}>
                            <FormattedMessage id="app.createCustomer" defaultMessage="Create Customer"/></Text>
                    </Col>
                    <Col size={10}/>
                    <Col size={1}>
                        <TouchableOpacity onPress={() => Actions.pop()}
                                          style={[styles.timesIcon, {marginTop: 0, backgroundColor: '#53535a'}]}>
                            <Icon size={scaleText(18)} name="close" color="white"/>
                        </TouchableOpacity>

                    </Col>
                </Row>
                {!this.state.customerCreated ?
                <Row size={6}>
                    <Col>
                        <Row style={{paddingLeft: scale(100), paddingRight: scale(100)}}>
                            <Col size={2}>
                                <Sae
                                    label={'First Name'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._firstName = el}
                                    onChangeText={(text) => {
                                        this.setState({firstName: text})
                                    }}
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                            <Col/>
                            <Col size={2}>
                                <Sae
                                    label={'Last Name'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._lastName = el}
                                    onChangeText={(text) => {
                                        this.setState({lastName: text})
                                    }}
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                        </Row>
                        <Row style={{paddingLeft: scale(100), paddingRight: scale(100)}}>
                            <Col size={2}>
                                <Sae
                                    label={'Address'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._address = el}
                                    onChangeText={(text) => {
                                        this.setState({address: text})
                                    }}
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                            <Col/>
                            <Col size={2}>
                                <Sae
                                    label={'City'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._city = el}
                                    onChangeText={(text) => {
                                        this.setState({city: text})
                                    }}
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                        </Row>
                        <Row style={{paddingLeft: scale(100), paddingRight: scale(100)}}>
                            <Col size={2}>
                                <Sae
                                    label={'Zipcode'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._zipcode = el}
                                    onChangeText={(text) => {
                                        this.setState({zipcode: text})
                                    }}
                                    keyboardType="numeric"
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                            <Col/>
                            <Col size={2}>
                                <Sae
                                    label={'Account Number'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._accountNumber = el}
                                    onChangeText={(text) => {
                                        this.setState({accountNumber: text})
                                    }}
                                    keyboardType="numeric"
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                        </Row>
                        <Row style={{paddingLeft: scale(100), paddingRight: scale(100)}}>
                            <Col size={2}>
                                <Sae
                                    label={'Mail'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._mail = el}
                                    onChangeText={(text) => {
                                        this.setState({mail: text})
                                    }}
                                    keyboardType="email-address"
                                    autoCapitalize={'none'}
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                            <Col/>
                            <Col size={2}>
                                <Sae
                                    label={'Phone'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._phone = el}
                                    onChangeText={(text) => {
                                        this.setState({phone: text})
                                    }}
                                    keyboardType="phone-pad"
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                        </Row>
                        <Row style={{paddingLeft: scale(100), paddingRight: scale(100)}}>
                            <Col>
                                <Sae
                                    label={'Notes'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._notes = el}
                                    onChangeText={(text) => {
                                        this.setState({notes: text})
                                    }}
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                        </Row>
                        <Row style={{paddingLeft: scale(100), paddingRight: scale(100)}}>
                            <Col>
                                <Row size={1}>
                                    <Text style={[styles.text, {
                                        backgroundColor: 'transparent',
                                        fontFamily: 'Gotham-Bold',
                                        color: '#7771ab',
                                        fontSize: scaleText(18)
                                    }]}><FormattedMessage id="app.groups" defaultMessage="Groups" /></Text>
                                </Row>
                                <Row size={3}>
                                    <SelectMultiple
                                        items={this.state.groups}
                                        selectedItems={this.state.customerGroups}
                                        onSelectionsChange={this.onSelectionsChange}
                                        labelStyle={[styles.text, {color: '#7771ab'}]}
                                        rowStyle={{backgroundColor: 'transparent', borderBottomWidth: 0, padding: 0}}
                                    />
                                </Row>
                            </Col>
                        </Row>
                        <Row size={0.5} style={{paddingLeft: scale(100), paddingRight: scale(100)}}>
                            <Col style={{paddingLeft: scale(200), paddingRight: scale(200)}}>
                                <ButtonItem text="app.okUpper" onPress={this.okOnPress}
                                            style={[styles.button, {backgroundColor: '#2dab61'}]}/>
                            </Col>
                        </Row>
                    </Col>
                </Row>: <View/>
                }
                {this.state.customerCreated ?
                <Row size={6}>
                    <Col>
                        <Row style={{paddingLeft: scale(100), paddingRight: scale(100)}}>
                            <Col size={2}>
                                <Sae
                                    label={'Name'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._accountName = el}
                                    onChangeText={(text) => {
                                        this.setState({accountName: text})
                                    }}
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                            <Col/>
                            <Col size={2}>
                                <Sae
                                    label={'Voucher Code'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._accountVoucher = el}
                                    onChangeText={(text) => {
                                        this.setState({accountVoucher: text})
                                    }}
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                        </Row>
                        <Row style={{paddingLeft: scale(100), paddingRight: scale(100)}}>
                            <Col size={2}>
                                <Sae
                                    label={'Card ID'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._accountCardID = el}
                                    onChangeText={(text) => {
                                        this.setState({accountCardID: text})
                                    }}
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                            <Col/>
                            <Col size={2}>
                                <Sae
                                    label={'Card Type'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._accountCardType = el}
                                    onChangeText={(text) => {
                                        this.setState({accountCardType: text})
                                    }}
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                        </Row>
                        <Row style={{paddingLeft: scale(100), paddingRight: scale(100)}}>
                            <Col size={2}>
                                <Sae
                                    label={'Credit'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._accountCredit = el}
                                    onChangeText={(text) => {
                                        this.setState({accountCredit: text})
                                    }}
                                    keyboardType="numeric"
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                            <Col/>
                            <Col size={2}>
                                <Sae
                                    label={'Balance'}
                                    iconClass={FontAwesomeIcon}
                                    iconName={'pencil'}
                                    iconColor={'white'}
                                    autoCorrect={false}
                                    style={{height: scale(16)}}
                                    ref={el => this._accountBalance = el}
                                    onChangeText={(text) => {
                                        this.setState({accountBalance: text})
                                    }}
                                    keyboardType="numeric"
                                    labelStyle={{fontFamily: 'Gotham-Bold'}}
                                />
                            </Col>
                        </Row>
                        <Row style={{paddingLeft: scale(100), paddingRight: scale(100)}}>
                            <Col size={2}>
                                <Row size={1}>
                                    <Text style={[styles.text, {
                                        backgroundColor: 'transparent',
                                        fontFamily: 'Gotham-Bold',
                                        color: '#7771ab',
                                        fontSize: scaleText(18)
                                    }]}><FormattedMessage id="app.accountType" defaultMessage="Account Type" /></Text>
                                </Row>
                                <Row size={3}>
                                    <SelectMultiple
                                        items={this.state.accountTypes}
                                        selectedItems={this.state.accountType}
                                        onSelectionsChange={this.onAccountTypeChange}
                                        labelStyle={[styles.text, {color: '#7771ab'}]}
                                        rowStyle={{
                                            backgroundColor: 'transparent',
                                            borderBottomWidth: 0,
                                            padding: 0
                                        }}
                                    />
                                </Row>
                            </Col>
                            <Col/>
                            <Col size={2}/>
                        </Row>
                        <Row size={0.5} style={{paddingLeft: scale(100), paddingRight: scale(100)}}>
                            <Col style={{paddingLeft: scale(200), paddingRight: scale(200)}}>
                                <ButtonItem text="app.okUpper" onPress={this.okOnPress}
                                            style={[styles.button, {backgroundColor: '#2dab61'}]}/>
                            </Col>
                        </Row>
                    </Col>
                </Row>: <View/>
                }
                <Row size={1}/>
                <AlertPopup ref={elt => this._alert = elt} intl={this.props.intl}/>
            </Grid>
        )
    }
}

export default connect(({app}) => ({
    departmentId: app.departmentId,
}))(CreateCustomerScreen)