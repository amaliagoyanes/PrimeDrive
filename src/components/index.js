export {default as Layout} from './Layout.js';
export {default as Login} from './Login.js';
export {default as MainScreen} from './MainScreen.js';
export {default as ReceiptsScreen} from './ReceiptsScreen.js';
export {default as TicketBlock} from './TicketBlock.js';
export {default as SideMenu} from './SideMenu.js';
export {default as TableManagement} from './TableManagement.js';
export {default as CreateCustomerScreen} from './CreateCustomerScreen.js';
export {default as CustomersScreen} from './CustomersScreen.js';
export {default as UserLoginScreen} from './UserLoginScreen.js';

export {default as ActionConfirm} from './popups/ActionConfirm.js';
export {default as AlertPopup} from './popups/AlertPopup.js';
export {default as CheckInOut} from './popups/CheckInOut.js';
export {default as CloseDay} from './popups/CloseDay.js';
export {default as NumberPopup} from './popups/NumberPopup.js';
export {default as OpenPricePopup} from './popups/OpenPricePopup.js';
export {default as OpenDay} from './popups/OpenDay.js';
export {default as Payment} from './popups/Payment.js';
export {default as SubProducts} from './popups/SubProducts.js';
export {SubProductsModal} from './popups/SubProducts.js';
export {default as TicketLine} from './popups/TicketLine.js';
export {default as UserLogin} from './popups/UserLogin.js';
export {default as ReceiptModal} from './popups/Receipt.js';
export {default as TableNameChange} from './popups/TableNameChange.js';
export {default as TablePicker} from './popups/TablePicker.js';
export {default as SelectRefundUnits} from './popups/SelectRefundUnits.js';
export {default as PrinterStatus} from './popups/PrinterStatus.js';
export {default as TerminalStatus} from './popups/TerminalStatus.js';
export {default as NetworkPrinterStatus} from './popups/NetworkPrinterStatus.js';
export {default as Scanner} from './popups/Scanner.js';
export {default as ChipCreator} from './popups/ChipCreator.js';
export {default as SplitPayment} from './popups/SplitPayment.js';

export {default as UserLoginBlock} from './popups/UserLoginBlock.js';
export {default as UserSearchBlock} from './popups/UserSearchBlock.js';

export * from './plain/index.js';