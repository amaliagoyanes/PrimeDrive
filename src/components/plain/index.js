export {default as NumericKeyboard} from './NumericKeyboard.js';
export {default as CashButton} from './CashButton.js';
export {default as ListViewRow} from './ListViewRow.js';
export {default as StatusSync} from './StatusSync.js';
export {default as ReceiptListViewRow} from './ReceiptListViewRow.js';
export {default as TablesLayout} from './TablesLayout.js';
export {default as Timer} from './Timer.js';

export * from './buttons/plainComponents.js';
export * from './modals/plainComponents.js';
export * from './receipts/plainComponents.js';