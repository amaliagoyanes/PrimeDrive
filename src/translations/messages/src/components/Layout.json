[
  {
    "id": "app.networkError",
    "defaultMessage": "Error occurred during loading data from server. Please check your network connection or try again later."
  },
  {
    "id": "app.noShiftWarning",
    "defaultMessage": "Please open the day first"
  }
]