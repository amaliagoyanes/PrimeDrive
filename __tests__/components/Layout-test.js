'use strict';

import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import {Layout} from '../../src/components/Layout';


Layout.prototype.componentDidMount = jest.fn;

let state = {
    pages: {page_1: {name: 'page_1'}},
    buttons: {button_1: {name: 'button_1', product: 'product_1', gridNumber: 1, location_x: 1, location_y: 1}},
    products: {product_1: {name: 'product_1', price: 2}},
};

it('renders correctly', () => {
    const tree = shallow(
        <Layout />);
    expect(tree).toMatchSnapshot();
    expect(tree.find('withAnimation(ProgressCircle)').length).toEqual(1);

});

it('renders correctly with state', () => {
    const tree = shallow(
        <Layout />);
    tree.setState(state);
    expect(tree).toMatchSnapshot();
});


it('renders correctly with state (open day) and layouts name', () => {
    const tree = shallow(
        <Layout/>);
    tree.setState(state);
    tree.setState({layouts: {name: "Test layout"}, timestamp: 1483228800});
    expect(tree).toMatchSnapshot();
    expect(tree.find('SwipeALot').length).toEqual(1);
    expect(tree.find('TicketBlock').length).toEqual(1);
    expect(tree.find('NumberPopup').length).toEqual(1);
});

it('renders correctly with state (open day), layouts name and cash shift id', () => {
    const tree = shallow(
        <Layout cash_shift_id={1}/>);
    tree.setState(state);
    tree.setState({layouts: {name: "Test layout"}, timestamp: 1483228800});
    expect(tree).toMatchSnapshot();
    expect(tree.find('SwipeALot').length).toEqual(2);
    expect(tree.find('TicketBlock').length).toEqual(1);
    expect(tree.find('NumberPopup').length).toEqual(1);
});