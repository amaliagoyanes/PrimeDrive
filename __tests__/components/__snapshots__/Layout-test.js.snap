// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <withAnimation(ProgressCircle)
      animated={true}
      borderColor="#2dab61"
      borderWidth={5}
      indeterminate={true}
      progress={0}
      size={60}
    />
  </View>
</View>
`;

exports[`renders correctly with state (open day) and layouts name 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <RowNB>
    <ColumnNB
      size={6}
    >
      <RowNB
        size={1}
      >
        <SwipeALot
          circleActiveStyle={
            Object {
              "backgroundColor": "#beb3b7",
            }
          }
          circleDefaultStyle={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#beb3b7",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 8,
              "margin": 7,
              "marginTop": 3,
              "top": -2,
              "width": 8,
            }
          }
        >
          <RowNB
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "marginBottom": 15,
              }
            }
          >
            <ButtonItem
              active={false}
              onPress={[Function]}
              text="page_1"
            />
            <ColumnNB />
            <ColumnNB />
          </RowNB>
        </SwipeALot>
      </RowNB>
      <RowNB
        size={5}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontFamily": "Gotham-Book",
                "fontSize": 14,
              }
            }
          >
            Please open the day first
          </Text>
        </View>
      </RowNB>
    </ColumnNB>
    <ColumnNB
      size={4}
    >
      <TicketBlock
        clearDataSource={[Function]}
        closeTicket={[Function]}
        dataSource={Array []}
        deleteItem={[Function]}
        deleteTicket={[Function]}
        number={0}
        onOk={[Function]}
        onPress={[Function]}
        parkTicket={[Function]}
        reduceDataSource={[Function]}
        saveTicketDiscount={[Function]}
        selectTable={[Function]}
        selectedTable={null}
        ticketData={Object {}}
        timestamp={1483228800}
        userData={null}
      />
    </ColumnNB>
    <NumberPopup
      arrow="left"
      deleteAll={true}
      onOk={[Function]}
      postfix="X"
      readOnly={true}
      showDelimeter={true}
    />
  </RowNB>
</View>
`;

exports[`renders correctly with state (open day), layouts name and cash shift id 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <RowNB>
    <ColumnNB
      size={6}
    >
      <RowNB
        size={1}
      >
        <SwipeALot
          circleActiveStyle={
            Object {
              "backgroundColor": "#beb3b7",
            }
          }
          circleDefaultStyle={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#beb3b7",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 8,
              "margin": 7,
              "marginTop": 3,
              "top": -2,
              "width": 8,
            }
          }
        >
          <RowNB
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "marginBottom": 15,
              }
            }
          >
            <ButtonItem
              active={false}
              onPress={[Function]}
              text="page_1"
            />
            <ColumnNB />
            <ColumnNB />
          </RowNB>
        </SwipeALot>
      </RowNB>
      <RowNB
        size={5}
      >
        <SwipeALot
          circleActiveStyle={
            Object {
              "backgroundColor": "#beb3b7",
            }
          }
          circleDefaultStyle={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#beb3b7",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 8,
              "margin": 7,
              "marginTop": 3,
              "top": -2,
              "width": 8,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "marginBottom": 15,
              }
            }
          >
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "marginBottom": 15,
              }
            }
          >
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "marginBottom": 15,
              }
            }
          >
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
            <RowNB>
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
              <ColumnNB />
            </RowNB>
          </View>
        </SwipeALot>
      </RowNB>
    </ColumnNB>
    <ColumnNB
      size={4}
    >
      <TicketBlock
        cash_shift_id={1}
        clearDataSource={[Function]}
        closeTicket={[Function]}
        dataSource={Array []}
        deleteItem={[Function]}
        deleteTicket={[Function]}
        number={0}
        onOk={[Function]}
        onPress={[Function]}
        parkTicket={[Function]}
        reduceDataSource={[Function]}
        saveTicketDiscount={[Function]}
        selectTable={[Function]}
        selectedTable={null}
        ticketData={Object {}}
        timestamp={1483228800}
        userData={null}
      />
    </ColumnNB>
    <NumberPopup
      arrow="left"
      deleteAll={true}
      onOk={[Function]}
      postfix="X"
      readOnly={true}
      showDelimeter={true}
    />
  </RowNB>
</View>
`;

exports[`renders correctly with state 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <withAnimation(ProgressCircle)
      animated={true}
      borderColor="#2dab61"
      borderWidth={5}
      indeterminate={true}
      progress={0}
      size={60}
    />
  </View>
</View>
`;
